### ISSUE TITLE
Feature Request: make HTTP status accessible in error callback

### ISSUE THREAD

**Issue by matthias-fratz-bsz** [openseadragon/openseadragon#2541]
In our use case, OpenSeadragon is running off a Cantaloupe IIIF server. We have Cantaloupe configured so it reports different HTTP status codes for the info.json, depending on what's wrong with the image ID. If the image ID was invalid or just doesn't exist, that's a simple 404. But if it is still being processed, we return a 504 instead. I would like to suggest the user what to do next depending on that status code: for 504, they can just reload the page in a few seconds; for 404, they need to fix their URL. For anything else, they might as well tell us immediately because it's probably on the Cantaloupe side. Another use case might be 401/403, telling the user to log in.

When calling addTiledImage(), we don't actually know whether the tileSource will be valid or not. It usually is, so I would like to avoid a round-trip just to check... but I would also like to display a helpful error message to the user if something does go wrong. Currently I would have to parse the `message` property to get the HTTP status, which doesn't feel right. Having the status accessible directly would be a much cleaner solution.

It would probably make sense to provide access to the `responseText` / `responseXML` as well, because that is commonly used for more detailed JSON error messages. I don't really know how to do that cleanly, though, when only one of them is really valid (I am not much of a JavaScript developer).

**Comment by iangilman** [comment]
Yes, that's a good idea! With the caveat that there can be failures that don't have an HTTP status associated with them. In those cases I guess we could just use `0`.

Looks like this would be a good place to start:

https://github.com/openseadragon/openseadragon/blob/fd74ea3dca9a414328312461e6f0f7c312ab9e11/src/tilesource.js#L560

... But there are probably a number of places along the chain that would need to be updated to make it clean.

If you'd like to work on this, please do! Otherwise hopefully someone will pick it up.

Btw, as you can see from that code, OSD is consistent about how it puts the status in the `message` so that should be a pretty reliable way to get the code in the meantime. I agree it's cleaner to have it be an explicit part of the event object, though!

**Comment by msalsbery** [comment]
@matthias-fratz-bsz 

Are you working on this? If not, I'll take it. I started but immediately noticed more <= IE11 code that could be removed so I'm going to do a patch for that first

**Comment by matthias-fratz-bsz** [comment]
@msalsbery No, I didn't find the time to work on it yet, so I'd be very happy if you do this one.

**Comment by msalsbery** [comment]
> @msalsbery No, I didn't find the time to work on it yet, so I'd be very happy if you do this one.

@matthias-fratz-bsz ok cool, thank you!
